pkg/predicate/toolchainconditionchanged.go

```diff
@@ -0,0 +1,66 @@
+package predicate
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var _ predicate.Predicate = (*ToolchainConditionChanged)(nil)
+
+type (
+	GetToolchainConditions func(client.Object) []toolchainv1alpha1.Condition
+
+	// ToolchainConditionChanged is a predicate to check that some conditions changed in a toolchain object.
+	ToolchainConditionChanged struct {
+		// GetConditions is function used to obtain the conditions from an object. There is no generic way of
+		// doing that using the standard k8s api.
```

Comment:
Yeah, I've thought about creating an interface on the top of all our API types that would enable us creating a generic API for the condition management. Right now, we technically re-do and copy-paste more or less the same code in all controllers. With an interface, we could have a set of generic functions, or a fluent API that could be reused without any other changes. Also the already existing functions in toolchain-common would work with the interface abstraction, not with the actual set of conditions.
But that's for later.